python/GafferSceneUI/InteractiveRenderUI.py

```diff
@@ -36,6 +36,70 @@
 
 import Gaffer
 import GafferScene
+import GafferUI
+
+##########################################################################
+# UI for the state plug that allows setting the state through buttons
+##########################################################################
+
+from collections import namedtuple
+
+ButtonLabels = namedtuple( 'ButtonLabels', 'startButton pauseButton')
+
+class InteractiveRenderingStateWidget( GafferUI.PlugValueWidget ) :
+
+	labelMap = {
+		GafferScene.InteractiveRender.State.Stopped : ButtonLabels('Start', 'Pause'),
+		GafferScene.InteractiveRender.State.Running : ButtonLabels('Stop', 'Pause'),
+		GafferScene.InteractiveRender.State.Paused : ButtonLabels('Continue', 'Paused') }
+
+	def __init__( self, plug, *args, **kwargs) :
+
+		self.plug = plug
+
+		row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+		GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+		with row :
+			self.startButton = GafferUI.Button()
+			self.pauseButton = GafferUI.Button()
+
+			self.startConnection = self.startButton.clickedSignal().connect( Gaffer.WeakMethod( self._startStopCallback ) )
+			self.pauseConnection = self.pauseButton.clickedSignal().connect( Gaffer.WeakMethod( self._pauseCallback ) )
+
+		self._updateButtons()
+
+	def _updateFromPlug( self ) :
+		pass
+
+	def hasLabel( self ) :
+		return True
+
+	def _isRunning( self ) :
+		return self.plug.getValue() == GafferScene.InteractiveRender.State.Running
+
+	def _isStopped( self ) :
+		return self.plug.getValue() == GafferScene.InteractiveRender.State.Stopped
```

Comment:
Generally, accessing the value of a plug without using a context is an alarm bell - if the plug is driven by an expression that requires specific context variables from the script, everything fails. So all plug accesses need to follow this pattern :

```
with self.getContext() :
   value = self.getPlug().getValue()
```